packages/react-button/src/components/Button/Button.types.ts

```diff
@@ -47,7 +47,7 @@ export type ButtonCommons = {
    * A button can format its icon to appear before or after its content.
    * @default 'before'
    */
-  iconPosition: 'before' | 'after';
+  iconPosition?: 'before' | 'after';
```

Comment:
Is this change correct? Looks a bit suspicious 😀

Reply:
It is to fix a compile error where MenuButton is casting from `MenuButtonState` to `ButtonState`:
```
src/components/MenuButton/useMenuButtonStyles.ts:36:19 - error TS2352: Conversion of type 'MenuButtonState' to type 'ButtonState' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'iconPosition' is missing in type 'MenuButtonState' but required in type 'ButtonCommons'.

36   useButtonStyles(state as ButtonState);
                     ~~~~~~~~~~~~~~~~~~~~

  src/components/Button/Button.types.ts:50:3
    50   iconPosition: 'before' | 'after';
         ~~~~~~~~~~~~
    'iconPosition' is declared here.
```


The error text makes sense, and seems like a legitimate error: MenuButtonState explicitly omits the `iconPosition` prop, so if it's required in ButtonState, then MenuButtonState can't be cast to ButtonState. My change to make `state.components` be required caused this to be reported as an error, but it shouldn't have anything to do with `iconPosition`... I don't know why it was not an error before this change. 

None of the existing Button code actually expected `iconPosition` to be defined, so it worked to simply make it optional. There are certainly other possible fixes here, but this one didn't change any runtime code, and seemed like a reasonable fix.